daffodil-core/src/main/scala/org/apache/daffodil/dsom/InitiatedTerminatedMixin.scala

```diff
@@ -46,8 +46,6 @@ trait InitiatedTerminatedMixin
    */
   lazy val hasInitiator = {
     val hasOne = initiatorExpr.isKnownNonEmpty
-    if (parentSaysInitiatedContent)
-      schemaDefinitionUnless(hasOne, "Enclosing group has initiatedContent='yes', but initiator is not defined.")
     hasOne
```

Comment:
I was hoping we could completely remove parentSaysInitiatedContent, but there's one place where it's still used:
```scala
lazy val initiatorDiscriminator = prod("initiatorDiscriminator", parentSaysInitiatedContent) { InitiatedContent(this) }
```
Any idea how we'll eventually remove thsi, since whether or not ``InitiatedContent`` is laid down really does depend on information about the parent. For example:
```xml
<xs:element name="foo" dfdl:initiator="foo" type="xs:string" />

<xs:element name="root">
  <xs:complexType>
    <xs:sequence dfdl:initiatedContent="yes">
      <xs:element ref="foo" />
    </xs:sequence>
    <xs:sequence dfdl:initiatedContent="no">
      <xs:element ref="foo" />
    </xs:sequence>
  </xs:complexType>
</xs:element>
```
In this case, a different grammar really does need to be laid down based on the context of where the foo element is reference. Any thoughts into how we'll handle something like this? Will the sequence pass down information to it's children so they have that content? Or the foo element have a reference to both enclosing sequences, and it will be passed an index to know which one is the current context or something?

Reply:
I'm pretty sure to fix this it has to happen in sequence child (or choice child or a base they both share). In generating the grammar or parser/unparser for that sequence child, that's where all the context-specific logic belongs. At that point the context is clear, and can be passed directly to anything that needs to make a decision based on it. The element ref or local element decl or child-term itself is then free of context-specific computations, and just provides information to the surrounding context.

Reply:
I also will set out to remove this particular context parent reference, by way of showing the new way things have to be done as code evolves to not use context pointers.

Reply:
I have concluded that this change is bigger and should be done in a separate change set and review. So not including this change as yet.